Makefile

```diff
@@ -1,50 +1,76 @@
 
 build-nofeatures: ## Build the container without http requests.
-	docker build -t $(DOCKER_REPO)/$(APP_NAME)_nofeatures:$(VERSION) $(COMMON_DOCKER_PARAMS) --build-arg CARGO_PARAMS="--no-default-features" .
+	docker build -t $(DOCKER_REMOTE_REPO)/$(APP_NAME)_nofeatures:$(VERSION) $(COMMON_DOCKER_PARAMS) --build-arg CARGO_PARAMS="--no-default-features" .
 
 run: ## Run container on port configured in `config.env`
 	docker run -it --rm --env HORUST_LOG=debug -v $(REPO_HOME)/examples/services/longrunning/:/etc/horust/services/ --name="$(NAME)" $(NAME):$(VERSION)
 
 run-bash: ## Run bash with horust
-	docker run -it --rm --env HORUST_LOG=debug --name="$(NAME)" $(DOCKER_REPO)/$(APP_NAME):$(VERSION) -- /bin/bash
+	docker run -it --rm --env HORUST_LOG=debug --name="$(NAME)" $(DOCKER_REMOTE_REPO)/$(APP_NAME):$(VERSION) -- /bin/bash
 
 stop: ## Stop and remove a running container
 	docker stop $(NAME)
 
-# Docker publish
+# Docker publishing tasks
 publish: build publish-latest publish-version ## publish the `{version}` ans `latest` tagged containers to ECR
 
 publish-latest: tag-latest ## publish the `latest` tagged container
-	@echo 'publish latest to $(DOCKER_REPO)'
-	docker push $(DOCKER_REPO)/$(APP_NAME):latest
+	@echo 'publish latest to $(DOCKER_REMOTE_REPO)'
+	docker push $(DOCKER_REMOTE_REPO)/$(APP_NAME):latest
 
 publish-version: ## publish the `{version}` tagged container to ECR
-	@echo 'publish $(VERSION) to $(DOCKER_REPO)'
-	docker push $(DOCKER_REPO)/$(APP_NAME):$(VERSION)
+	@echo 'publish $(VERSION) to $(DOCKER_REMOTE_REPO)'
+	docker push $(DOCKER_REMOTE_REPO)/$(APP_NAME):$(VERSION)
 
-tag-latest:
+tag-latest: ## tags the latest container with the version listed above
 	@echo 'create tag latest'
-	docker tag $(DOCKER_REPO)/$(APP_NAME):$(VERSION) $(DOCKER_REPO)/$(APP_NAME):latest
+	docker tag $(DOCKER_REMOTE_REPO)/$(APP_NAME):$(VERSION) $(DOCKER_REMOTE_REPO)/$(APP_NAME):latest
 
 version: ## output to version
-	@echo $(VERSION)
\ No newline at end of file
+	@echo ''$(VERSION)
+
+# Docker local development tasks
+## Dargo == Docker Cargo
+dargo-prep: ## This runs everything neccessary to start developing locally in a container
+	# Spin up a long-running rust container
+	make dargo-create-container
+	# Compile and cache all dependencies
+	make dargo COMMAND=build
+	# Perform a first-sweep check, to fill up the cache (it seems to be different than the build one)
+	make dargo COMMAND=check
+    # Compile and cache test dependencies, run tests
+	make dargo COMMAND=test
+	# Now all caches are filled, and subsequent operations will be faster
+	# Consider adding dargo(){ make dargo COMMAND=$1} to your ~/.zshrc or ~/.bashrc for ergonomics
+
+dargo: ## Run a cargo command inside the container
+	docker exec -ti docker-horust cargo $(COMMAND)
+
+dargo-create-container: ## Create a Rust container with this folder bind-mounted to it
+	@echo 'building rust image for local development'
+	docker build -t horust -f localdev/Dockerfile localdev/
```

Comment:
You can skip this step because we can just use rust:1.42 as image instead of this one.

Reply:
But I'd like to set the `WORKDIR` to `/usr/src/Horust` instead of `/` and have the `CMD` set to `/bin/bash`. That `localdev/Dockerfile` now is just:


```Dockerfile
FROM rust:1.42
WORKDIR /usr/src/Horust
CMD ["/bin/bash"]
```

If you look [here](https://github.com/rust-lang/docker-rust/blob/8bab191937fcf23569d3a3c31103c1c6f7f2947e/1.42.0/buster/Dockerfile), the original `rust:1.42` Dockerfile does neither.

Or am I missing something?

Reply:
You can pass those as parameters of docker run. https://docs.docker.com/engine/reference/run/#workdir
So docker run -d rust:142 -w .... /bin/bash

Reply:
True, testing now

Reply:
Only flags now, deleting the localdev folder